doc/nrf/app_dev/device_guides/nrf54l/building_nrf54l.rst

```diff
@@ -0,0 +1,129 @@
+
+Depending on the sample, you must program only the application core or both the Fast Lightweight Peripheral Processor (FLPR) and the application core.
+Additionally, the process will differ based on whether you are working with a single-image or multi-image build.
+
+Building for the application core
+*********************************
+
+For instructions on building for the application core only, see how to :ref:`program an application <programming_cmd>`.
+
+Building for the application and FLPR core
+******************************************
+
+This section outlines how to build and program for both the application and FLPR core, covering separate builds and sysbuild configurations.
+FLPR core supports two variants:
+
+* ``nrf54l15dk/nrf54l15/cpuflpr`` where FLPR runs from SRAM (recommended).
+   To build FLPR image with this variant, the application core image must include the ``nordic-flpr`` snippet.
+
+* ``nrf54l15dk/nrf54l15/cpuflpr/xip`` where FLPR runs from RRAM.
+   To build FLPR image with this variant, the application core image must include the ``nordic-flpr-xip`` snippet.
+
+Separate images
+---------------
+
+To build and program the application sample and the FLPR sample as separate images, follow these steps:
+
+.. tabs::
+
+   .. group-tab:: west
+
+      1. |open_terminal_window_with_environment|
+      #. Build the application core image, and based on your build target include the appropriate snippet:
+
+         .. code-block:: console
+
+            west build -p -b nrf54l15dk/nrf54l15/cpuapp -S nordic-flpr --no-sysbuild
+
+      #. Flash the application core image by running the following command:
+
+         .. code-block:: console
+
+            west flash
+
+      #. Build the FLPR core image:
+
+         .. code-block:: console
+
+            west build -p -b nrf54l15dk/nrf54l15/cpuflpr --no-sysbuild
+
+      #. Once you have sucessfully built the FLPR core image, flash it onto your device:
+
+         .. code-block:: console
+
+            west flash
+
+   .. group-tab:: nRF Util
+
+      1. |open_terminal_window_with_environment|
+      #. For the FLPR core, run the following command to erase the flash memory of the FLPR core and program the sample:
+
+         .. Commands TBD
+
+         .. code-block:: console
+
+            nrfutil device program --firmware zephyr.hex --options chip_erase_mode=ERASE_ALL --core Network
+
+      #. For the application core, navigate to its build folder and run:
+
+         .. code-block:: console
+
+            nrfutil device program --firmware zephyr.hex  --options chip_erase_mode=ERASE_ALL
+
+         .. note::
+            The application build folder will be in a sub-directory which is the name of the folder of the application
+
+      #. Reset the development kit:
+
+         .. Commands TBD
+
+         .. code-block:: console
+
+            nrfutil device reset --reset-kind=RESET_PIN
+
+         See :ref:`readback_protection_error` if you encounter an error.
+
+With sysbuild
```

Comment:
I'm missing an instruction how to build an application image that uses sysbuild to include FLPR as sub-image. I think that's the most important scenario for application developers

Reply:
Perhaps this should be added when we have an example application for such usecase?

My expectation would be that to do that you need to:
- add contents of `nordic-flpr-*` overlay to app devicetree
- add sub-image with FLPR target to sysbuild
- build for app target
- flash will cover both targets

Reply:
It's fine for me to add this later. Probably it would be good to explicitly state that this usage scenario is not documented yet and for now, we recommend using FLPR as the main CPU in multi-core projects.

At least I, as an application developer, would be surprised by the suggestion of using FLPR as the main CPU and I would be looking for other options

Reply:
Added a note, please double-check (line 42), switched the order of these sections as per Marcin's suggestions

Reply:
Continuing this discussion in a comment on the note: https://github.com/nrfconnect/sdk-nrf/pull/17443#discussion_r1827439901